services/QuillLMS/app/queries/snapshots/average_query.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Snapshots
+  class AverageQuery < PeriodQuery
+    attr_reader :params
+
+    def initialize(*params)
+      # We pull params here because we're going to do a naive pass-through to sibling queries below
+      @params = params
+
+      super
+    end
+
+    # Adding a no-op overwrite of query to avoid confusion in debugging where this class
+    # might assemble a query even though it never intends to run one because it pulls data
+    # from other queries instead
+    def query
```

Comment:
You're right to call out this difference. This class's behavior is different enough from the others that I'd like to consider inheriting from a new parent class, `ComposedQuery`, which has no `query` method.

Reply:
I concur with Peter's suggestion of extracting out a new `ComposeQuery` class.